calendar/client.bal

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+
+# Ballerina Microsoft Outlook Calendar connector provides the capability to access MS Outlook Calendar API.
+# This connector provides the capability to easily create appointments & events, organize meetings, 
+# manage user's calendar etc.
+#
+# + httpClient - the HTTP Client
+@display {
+    label: "Microsoft Calendar Client",
+    iconPath: "logo.svg"
+}
+public isolated client class Client {
+    private final http:Client httpClient;
+    final readonly & Configuration config;
+
+    # Initializes the connector. 
+    # During initialization you can pass either http:BearerTokenConfig
```

Comment:
To initialize, you can either pass the http:BearerTokenConfig if you have a bearer token or the http:OAuth2RefreshTokenGrantConfig if you have OAuth 2.0 tokens.

Reply:
Hi Shani, 
This Client configuration record has been unified across connectors in the new release [1]. 
This record type accepts several authentication modes [2].

[1] https://github.com/ballerina-platform/module-ballerinax-microsoft.outlook.calendar/blob/4e2c2ad018816ffe0a3fee9c205f988472f2afe0/calendar/client.bal#L31
[2] https://github.com/ballerina-platform/module-ballerinax-microsoft.outlook.calendar/blob/4e2c2ad018816ffe0a3fee9c205f988472f2afe0/calendar/types.bal#L21

Thanks